crates/electrum/src/bdk_electrum_client.rs

```diff
@@ -249,16 +257,17 @@ impl<E: ElectrumApi> BdkElectrumClient<E> {
         mut spks_with_expected_txids: impl Iterator<Item = (u32, SpkWithExpectedTxids)>,
         stop_gap: usize,
         batch_size: usize,
+        pending_anchors: &mut Vec<(Txid, usize)>,
     ) -> Result<Option<u32>, Error> {
-        let mut unused_spk_count = 0_usize;
-        let mut last_active_index = Option::<u32>::None;
+        let mut unused_spk_count = 0;
+        let mut last_active_index = None;
 
-        loop {
+        'batch_loop: loop {
             let spks = (0..batch_size)
                 .map_while(|_| spks_with_expected_txids.next())
                 .collect::<Vec<_>>();
             if spks.is_empty() {
-                return Ok(last_active_index);
+                break;
```

Comment:
Is this set of changes intended to improve readability? I think it is doing the opposite effect.

* `0_usize` is clearer than `0` as it contains the type and the reader is not required to rely on the LSP.
* `Option::<u32>::None` is clear than `None`.
* `return Ok(last_active_index)` is clearer than `break` as it is more linear to read. `break` means we need to follow where the loop ends.

I recommend reverting these changes.